byterun/gc_ctrl.c

```diff
@@ -633,16 +643,18 @@ extern int caml_parser_trace;
 
 CAMLprim value caml_runtime_parameters (value unit)
 {
+#define F_Z ARCH_INTNAT_PRINTF_FORMAT
+
   CAMLassert (unit == Val_unit);
   return caml_alloc_sprintf
-    ("a=%d,b=%s,H=%lu,i=%lu,l=%lu,o=%lu,O=%lu,p=%d,s=%lu,t=%d,v=%lu,w=%d,W=%lu",
-     /* a */ caml_allocation_policy,
+    ("a=%d,b=%d,H=%"F_Z"u,i=%"F_Z"u,l=%"F_Z"u,o=%"F_Z"u,O=%"F_Z"u,p=%d,s=%"F_Z"u,t=%"F_Z"u,v=%"F_Z"u,w=%d,W=%"F_Z"u",
```

Comment:
This is now on trunk as https://github.com/ocaml/ocaml/commit/70c78d5c4b2b2b202e3eb7a4d85e5f1b031ab403 and 4.04 as e4af3e2169184357af55930ed154f5cc04f2668d and will be on 4.05 as soon as the CI completes there.